operators/src/pro/ml/xgboost.rs

```diff
@@ -0,0 +1,628 @@
+use std::mem;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use futures::{future, stream, StreamExt, TryStreamExt};
+use geoengine_datatypes::hashmap;
+use geoengine_datatypes::primitives::{
+    ClassificationMeasurement, Measurement, RasterQueryRectangle, SpatialPartition2D,
+    SpatialResolution,
+};
+use geoengine_datatypes::raster::{
+    BaseTile, Grid2D, GridOrEmpty, GridShape, GridShapeAccess, GridSize, Pixel, RasterDataType,
+    RasterTile2D,
+};
+use rayon::prelude::ParallelIterator;
+use rayon::slice::ParallelSlice;
+use rayon::ThreadPool;
+use serde::{Deserialize, Serialize};
+use xgboost::{Booster, DMatrix};
+
+use crate::engine::{
+    CreateSpan, ExecutionContext, InitializedRasterOperator, MultipleRasterSources, Operator,
+    OperatorName, QueryContext, QueryProcessor, RasterOperator, RasterQueryProcessor,
+    RasterResultDescriptor, TypedRasterQueryProcessor,
+};
+use crate::error;
+use crate::util::stream_zip::StreamVectorZip;
+use crate::util::Result;
+use futures::stream::BoxStream;
+use RasterDataType::F32 as RasterOut;
+
+use tracing::{span, Level};
+use TypedRasterQueryProcessor::F32 as QueryProcessorOut;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct XgboostParams {
+    model_file_path: PathBuf,
+    no_data_value: Option<f32>,
+}
+
+pub type XgboostOperator = Operator<XgboostParams, MultipleRasterSources>;
+
+impl OperatorName for XgboostOperator {
+    const TYPE_NAME: &'static str = "XgboostOperator";
+}
+
+pub struct InitializedXgboostOperator {
+    result_descriptor: RasterResultDescriptor,
+    sources: Vec<Box<dyn InitializedRasterOperator>>,
+    model_file_path: PathBuf,
+    no_data_value: Option<f32>,
+}
+
+type PixelOut = f32;
+
+#[typetag::serde]
+#[async_trait]
+impl RasterOperator for XgboostOperator {
+    async fn _initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<dyn InitializedRasterOperator>> {
+        let init_rasters = future::try_join_all(
+            self.sources
+                .rasters
+                .iter()
+                .map(|raster| raster.clone().initialize(context)),
+        )
+        .await?;
+
+        let input = init_rasters.get(0).unwrap();
```

Comment:
you should throw an error like, XgBoost requires at least 1 raster input

Reply:
https://github.com/geo-engine/geoengine/commit/84fb951c583cb7141ff50f1196f90504d2ef8a90